lambdas/process_webhooks/process_webhooks.py

```diff
@@ -53,7 +63,257 @@ def _get_target_url(headers):
             "type: {}".format(event_type)
         )
 
-    return url + "/" + endpoint
+    return base_url + "/" + endpoint
+
+
+def _get_credentials_from_s3(jenkins_url):
+    """
+    Get jenkins credentials from s3 bucket.
+    The expected object is a JSON file formatted as:
+    {
+        "username": "sampleusername",
+        "api_token": "sampletoken"
+    }
+    """
+    # For both build and test jenkins, we can use
+    # the same credentials
+    session = botocore.session.get_session()
+    client = session.create_client('s3')
+
+    creds_file = client.get_object(
+        Bucket=CREDENTIALS_BUCKET,
+        Key=CREDENTIALS_FILE
+    )
+    creds = json.loads(creds_file['Body'].read())
+
+    if not creds.get("username") or not creds.get("api_token"):
+        raise StandardError(
+            'Credentials file needs both a '
+            'username and api_token attribute'
+        )
+
+    return creds["username"], creds["api_token"]
+
+
+def _get_jobs_list(repository, target):
+    """
+    Find the list of jobs that the webhook should kick
+    off on Jenkins.
+    """
+    jobs_list = []
+
+    # Based on the repo, target, and event type find the
+    # desired list of tests from constants.py
+    if repository == 'edx-platform':
+        if target == 'master':
```

Comment:
I am thinking through this, and there are cases in which we want to test PRs not aimed at master, for instance, collaboration on a long running pr or a patch. Perhaps we could check to see if it is targeting the special release branches first and just 'else' the rest of the targets for the 'edx-platform' repository